pvlib/bifacial/loss_models.py

```diff
@@ -0,0 +1,175 @@
+    output power:
+
+    .. math::
+
+       M[\%] = 1 - \frac{P_{Array}}{\sum P_{Cells}} \qquad \text{(1)}
+
+    To account for a module with a fill factor distinct from the one used to
+    train the model (0.79), the output of the model can be modified by Eq. (7):
+
+    .. _fillfactor_eq:
+
+    .. math::
+
+       M[\%]_{FF_1} = M[\%]_{FF_0} \frac{FF_1}{FF_0} \qquad \text{(7)}
+
+    In the section *See Also*, you will find two packages that can be used to
+    calculate the irradiance at different points of the module.
+
+    .. note::
+       The global irradiance RMAD is different from the backside irradiance
+       RMAD.
+
+    In case the RMAD of the backside irradiance is known, the global RMAD can
+    be calculated as follows, assuming the front irradiance RMAD is
+    negligible [2]_:
+
+    .. math::
+
+       RMAD(k \cdot X + c) = RMAD(X) \cdot k \frac{k \bar{X}}{k \bar{X} + c}
+       = RMAD(X) \cdot k \frac{1}{1 + \frac{c}{k \bar{X}}}
+
+    by similarity with equation (2) of [1]_:
+
+    .. math::
+
+       G_{total\,i} = G_{front\,i} + \phi_{Bifi} G_{rear\,i} \qquad \text{(2)}
+
+    See Also
+    --------
+    `solarfactors <https://github.com/pvlib/solarfactors/>`_
+        Calculate the irradiance at different points of the module.
+    `bifacial_radiance <https://github.com/NREL/bifacial_radiance>`_
+        Calculate the irradiance at different points of the module.
+
+    References
+    ----------
+    .. [1] C. Deline, S. Ayala Pelaez, S. MacAlpine, and C. Olalla, 'Estimating
+       and parameterizing mismatch power loss in bifacial photovoltaic
+       systems', Progress in Photovoltaics: Research and Applications, vol. 28,
+       no. 7, pp. 691-703, 2020, :doi:`10.1002/pip.3259`.
+    .. [2] “Mean absolute difference,” Wikipedia, Sep. 05, 2023.
+       https://en.wikipedia.org/wiki/Mean_absolute_difference#Relative_mean_absolute_difference
+       (accessed 2024-04-14).
+    """  # noqa: E501
+    if isinstance(model, str):
+        _MODEL_POLYNOMS = {
+            "fixed-tilt": [0, 0.142, 0.032],  # Eq. (11), [1]
+            "single-axis-tracking": [0, 0.054, 0.068],  # Eq. (12), [1]
+        }
+        try:
+            model_polynom = np.polynomial.Polynomial(_MODEL_POLYNOMS[model])
+        except KeyError:
+            raise ValueError(
+                f"Invalid model '{model}'. Available models are "
+                f"{list(_MODEL_POLYNOMS.keys())}."
+            )
+        else:
+            if fillfactor:
+                # Use fillfactor to modify output of a known trained model
+                # Eq. (7), [1]
+                model_polynom = model_polynom * fillfactor / 0.79
+    else:
+        if fillfactor:
+            raise ValueError(
+                "Fill factor can only be used with predefined models. "
+                "Modify polynomial or multiply output by "
+                "'module_fillfactor / training_fillfactor'."
```

Comment:
Why is this restriction necessary?  I'd have thought that the fill factor adjustment could be done for any set of coefficients.  Can't we just apply it at the end of the function after the polynomial evaluation?

Reply:
I may have pointed @echedey-ls in that direction. The model authors assert that mismatch loss is proportional to the fill factor / 0.79 (fill factor used to establish the built-in coefficients), so the fill factor scaling is part of the author's model. However, combining fill factor scaling with a user-supplied polynomials should be at least a Warning, since a user polynomial may not have been fit to values consistent with fill factor of 0.79.

Reply:
Right, of course.  Thanks.  I suppose we could have a `fill_factor_ref=0.79`, if we wanted to better support user-supplied coefficients.  I would also be fine with just a warning.

Reply:
I didn't think about that reference fill factor; I love that idea!